Recast/Source/RecastMesh.cpp

```diff
@@ -427,18 +427,16 @@ static int triangulate(int n, const int* verts, int* indices, int* tris)
 		for (int k = i1; k < n; k++)
 			indices[k] = indices[k+1];
 		
-		if (i1 >= n) i1 = 0;
-		i = prev(i1,n);
 		// Update diagonal flags.
-		if (diagonal(prev(i, n), i1, n, verts, indices))
-			indices[i] |= 0x80000000;
-		else
-			indices[i] &= 0x0fffffff;
-		
-		if (diagonal(i, next(i1, n), n, verts, indices))
-			indices[i1] |= 0x80000000;
-		else
-			indices[i1] &= 0x0fffffff;
+		for (int i = 0; i < n; i++)
+		{
+			int i1 = next(i, n);
+			int i2 = next(i1, n);
+			if (diagonal(i, i2, n, verts, indices))
+				indices[i1] |= 0x8000;
```

Comment:
Probably a small mixup with tilecache here and 2 lines down, shouldn't these feature the prevous constants (0x80000000 and 0x0fffffff)?

Reply:
Yes, my mistake. Sorry, I subconsciously assumed that the values of the two were the same

Reply:
I have submitted my modifications and thank you very much for pointing out my issues